docs/database/cosmosdb.md

```diff
@@ -451,5 +456,448 @@ It captures the number of active instances at any given time. Best practices dic
+| `db.client.cosmosdb.request.body.size` | Histogram | `By` | Size of request body sent over wire to Azure Cosmos DB backend service. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| [`db.collection.name`](/docs/attributes-registry/db.md) | string | Cosmos DB container name. [1] | `public.users`; `customers` | `Conditionally Required` If available | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.cosmosdb.consistency_level`](/docs/attributes-registry/db.md) | string | Effective Request [consistency level](https://learn.microsoft.com/azure/cosmos-db/consistency-levels). This represents the applicable consistency level determined based on the hierarchy of settings: request-level, client instance-level, and account-level default consistency levels. | `Eventual`; `ConsistentPrefix`; `BoundedStaleness`; `Strong`; `Session` | `Conditionally Required` If available. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.cosmosdb.network.response.status_code`](/docs/attributes-registry/db.md) | int | The status code returned in response by the network call, made to Cosmos DB backend services. [2] | `200` | `Conditionally Required` If available | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.cosmosdb.network.response.sub_status_code`](/docs/attributes-registry/db.md) | int | The Azure Cosmos DB sub-status code for a network request, made to Cosmos DB backend services, during an operation. | `1002` | `Conditionally Required` If available | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.cosmosdb.sub_status_code`](/docs/attributes-registry/db.md) | int | The Azure Cosmos DB sub-status code for an operation represents the final sub-status code after multiple network interactions have been completed during the operation. | `1000`; `1002` | `Conditionally Required` when response was received and contained sub-code. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.namespace`](/docs/attributes-registry/db.md) | string | The name of the database, fully qualified within the server address and port. | `customers`; `test.users` | `Conditionally Required` If available. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.operation.name`](/docs/attributes-registry/db.md) | string | The name of the operation or command being executed. [3] | `create_item`; `query_items`; `read_item` | `Conditionally Required` [4] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`db.response.status_code`](/docs/attributes-registry/db.md) | string | Database response status code. [5] | `200`; `201` | `Conditionally Required` [6] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
I still think it's highly problematic to request every instrumentation to:
- measure latency (with all attributes) at each network call
- propagate it back to the public API surface
- report these metrics there

This behavior is very difficult to implement is error prone. It would be problematic to implement in SDKs that don't get that much love such as Python, Go, JS, etc. Also I believe Java does not report metrics in this way now - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/docs/Metrics.md

It also skews analysis since metrics will be reported at a different time than when response is received. If there are some locks or cases when multiple logical operations are in flight, there could be periods of time when no metrics are reported (since everything is waiting).

Given this, even if you do this now, you should have an escape path that allows you to change this behavior in the future.

The suggestion is to either
- report `db.response.status_code` and `db.cosmosdb.sub_status_code` only and describe only the network side of things in each measurement. I'd prefer this path
OR
- make  `db.response.status_code` and `db.cosmosdb.sub_status_code` `recommended` so that SDKs like Python or Go won't be required to implement all this logic and .NET would be able to  change this behavior.

Reply:
I understand your concern, but we are leveraging existing design to generate the metrics and avoiding any new code in the hot path.
**This behavior is very difficult to implement is error prone** => We have diagnostic engine implemented in SDK which collects all the information for a request and can be used to generate diagnostic JSON for that request (on demand). So, this engine will provide the data point for the metrics.
**Java does not report metrics in this way** => Java report metrics in this way only. There also we use diagnostic engine to generate these metrics. @FabianMeiswinkel can you please confirm it.

I would go with 2nd option; it would help us to correlate/analyze the network level behavior of successful operations.

@kirankumarkolli @FabianMeiswinkel thoughts?

Reply:
My 2 cents - keep all transport-level metrics out-of semantic preofile. There simply is no need to standardize it - we intentionally disable transport-level metrics by default anyway - and the whole purpose of them is debugging certain situations. When we have the new connection mode thin client proxy that will be supported across all SDKs, we can reconsider this - because then we will have one set of transport metrics that is applicable to all SDKs
Right now - let .Net and Java do whatever fits their needs for transport metrics - trying to standardize this metric is simply over-engineering.

Reply:
> My 2 cents - keep all transport-level metrics out-of semantic preofile

If metrics are not formally documented, they would be implemented in different ways in different languages. Also, users won't be able to understand what these metrics report or build shareable dashboards.

> Java report metrics in this way only. There also we use diagnostic engine to generate these metrics. 

It does not list two different sets of status attributes (global and local)  though

If you feel strong that it's necessary to report global operation status on all network calls, please apply my second suggestion and make it recommended and not required: 

> make db.response.status_code and db.cosmosdb.sub_status_code recommended so that SDKs like Python or Go won't be required to implement all this logic and .NET would be able to change this behavior.